paddle/fluid/framework/ir/memory_optimize_pass/memory_reuse_pass.cc

```diff
@@ -131,14 +120,24 @@ void MemoryReusePass::CollectShareTensorBufferOpHandles() const {
 
 void MemoryReusePass::CollectReusedVars() const {
   for (auto &pair : ops_) {
-    auto reused_vars = pair.second->ReusedVarSet();
-    reused_var_names_[pair.first->GetScopeIdx()].insert(reused_vars.begin(),
-                                                        reused_vars.end());
+    auto reused_vars = pair.second->ReusedVars();
+    for (auto &reused_var_pair : reused_vars) {
+      reused_in_var_names_[pair.first->GetScopeIdx()].insert(
+          reused_var_pair.first);
+      reused_out_var_names_[pair.first->GetScopeIdx()].insert(
+          reused_var_pair.second);
+    }
   }
 }
 
-bool MemoryReusePass::IsVarAlreadyReused(details::VarHandle *var) const {
-  return reused_var_names_[var->scope_idx()].count(var->Name()) > 0;
+bool MemoryReusePass::IsInVarAlreadyReused(details::VarHandle *in_var) const {
```

Comment:
Maybe in future PR, because I saw a lot in this PR.

In general, there are some exceptions but input arguments are values or const references while output arguments are pointers.

Ref: https://google.github.io/styleguide/cppguide.html#Reference_Arguments

Reply:
Done.